_posts/2020-07-27-cpp-build-simplification.md

```diff
@@ -0,0 +1,271 @@
+---
+layout: post
+title: "INSERT TITLE HERE"
+date: "2020-07-24 00:00:00 -0600"
+author: pmc
+categories: [application]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+Over the last four and a half years, we've worked to build a
+"batteries-included" development platform for high-performance analytics
+applications in C++. As the scope of the project has grown, we have sometimes
+taken on additional library dependencies to support a wide variety of systems
+and data processing tasks.
+
+While these dependencies give us leverage on hard problems, in some cases their
+addition has caused extra complexity for projects that depend on Arrow, even if
+their use of Arrow is limited to basic interactions with the Arrow columnar
+format. Some projects have therefore been concerned about the implications of
+taking on the Arrow C++ library as a dependency if Arrow's build and runtime
+requirements spill over into their project's requirements. Indeed, in the
+earlier stages of the Arrow project development, dependency management issues
+did cause problems for early adopters.
+
+We want developers to trust that they can use and depend on our libraries, and
+that doing so doesn't add a burden for their own project maintenance or for
+their users. Over the last year, we have undertaken a number of significant
+projects to accommodate the different ways that people want to depend on Arrow
+C++. We've aimed to make the build process simple by default, without requiring
+special environment setup, yet also highly configurable for those who need to
+specialize. And we've worked to make builds faster and more compact, even as we
+continue to add new functionality.
```

Comment:
Maybe "This also includes a zero-dependency option for projects that wish to use the Arrow C++ core but take on no transitive dependencies."